src/theme/components/template.ts

```diff
@@ -13,77 +16,113 @@ const Template = defineStyleConfig({
       1fr
       min(1280px, 100%)
       1fr`,
-    rowGap: "grid.l",
+    rowGap: responsiveSpacing,
     _dark: {
       color: "dark.ui.typography.body",
     },
   }),
   sizes: {},
   defaultProps: {},
 });
-// Elements that need to breakout will span outside
-// the center 1280px grid column.
+/** Elements that need to breakout will span outside the center 1280px grid
+ * column. */
 const TemplateBreakout = defineStyleConfig({
   baseStyle: defineStyle({
     width: "100%",
-    // This could be "1 / 4" and it would mean the same. This is
-    // "future-proof" the grid column assignment to the last column.
+    /** This could be "1 / 4" and it would mean the same. This is "future-proof"
+     * the grid column assignment to the last column. */
     gridColumn: "1 / -1",
   }),
 });
 const TemplateContent = defineStyleConfig({
-  baseStyle: defineStyle({
-    // Set this element to start on the second 1280px grid column.
+  baseStyle: defineStyle(({ useLegacyGrid = false }) => ({
```

Comment:
Is there a reason why the checks have to happen in every style definition separately, rather than once up top (eg: switch vs if-else)

Reply:
The conditions are used to set specific values for various style attributes and we've found it is easier to see the condition right on the attribute rather than abstracting the condition into a separate function elsewhere in the file.

Reply:
Is it easier because Chakra favors using one pattern over the other?  Or is this an argument that an if-else ladder is an easier pattern to read in general?  Or an if-else ladder is easier to read in this specific situation because ... ?

Reply:
We've continued to use patterns that Chakra shows in their docs.  That being said, I am curious about the solution you are suggesting.  There are a number of styles for separate elements that are dependent on the `useLegacyGrid` prop.  What type of design pattern would you suggest to consolidate the conditions based on that prop value?

Reply:
This just comes from the principle that, if we're having a switch that we expect to remove in the future, it's easier to read and easier to remove if we define the entire legacy grid style in one place, and the entire new grid style in another place.  Now, if the tools we use don't allow that (I know Storybook doesn't, but I didn't know if Chakra didn't) then we don't have much of a choice.